packages/nimble-components/src/table/specs/table-keyboard-navigation-hld.md

```diff
@@ -0,0 +1,180 @@
+# Table Keyboard Navigation HLD
+
+## Problem Statement
+
+It is desirable for the table to support navigating and interacting with its contents with the keyboard alone. ARIA provides guidance on how to best accomplish this.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [Previous HLD and discussions on pull request](https://github.com/ni/nimble/pull/1506)
+-   [ARIA Treegrid Pattern](https://www.w3.org/WAI/ARIA/apg/patterns/treegrid/#keyboardinteraction)
+-   [ARIA Focus/Edit/Navigation guidance for cells](https://www.w3.org/WAI/ARIA/apg/patterns/grid/#gridNav_focus)
+-   [ARIA treegrid example](https://codepen.io/atmgrifter00/pen/oNJLMQr)
+-   Outdated:
+    -   [Prototype branch](https://github.com/ni/nimble/tree/table-keyboard-interactions-treegrid) ([Storybook](https://60e89457a987cf003efc0a5b-vtowbhsqmj.chromatic.com/?path=/story/incubating-table--table&args=selectionMode:multiple))
+    -   [IxD Keyboard interactions](https://xd.adobe.com/view/fa09e396-dbb9-40b8-547f-1cf9eab35a0b-8c38/)
+
+## Implementation / Design
+
+We have decided to use the ARIA `treegrid` role/ pattern for our table component (as opposed to the `grid` role).
+
+### ARIA Treegrid guidance
+
+Some noteworthy aspects of the treegrid pattern are:
+
+1. All cells are focusable, even those that have no interactive elements. The left and right arrow keys, while in navigation modes, move focus from cell to cell. In the case where a cell has a single interactive element, the element can be given the focus instead of the containing cell.
```

Comment:
>In the case where a cell has a single interactive element, the element can be given the focus instead of the containing cell.

Can you clarify this statement? I'm not sure if you mean
(a) The cell contains only one interactive element (but possibly other elements that are not interactive)
or
(b) The cell contains only one element, and that element is interactive

Based on the excerpt from the ARIA guidelines, I'm guessing you mean (b), but when I first read the sentence, I thought you meant (a).

Reply:
I was going to say (b), but I'm not sure that it's a hard & fast rule. If there's multiple interactive elements in the cell, we definitely won't try to focus any of them (just the cell).

But, for example, if a cell had a button or checkbox, plus some extra accompanying text (not the control label), it would probably still make sense to just focus the button/checkbox.

The approach I described in the nimble-table section below is actually closer to (a) too.

Reply:
Updated HLD wording (closer to your (a) option)

Reply:
I'm still a bit confused by this. The ARIA guidelines seem to be saying that the cell needs to be focusable in order for the non-focusable text to be read by a screen reader. However, the plan I'm reading in your statement is that neither the cell nor the non-focusable text will be focusable if the cell contains only one interactive element. Those statements seem in conflict with each other. Am I missing some aspect of what you are proposing here?

Reply:
I think our exact behavior here might depend on the implementation approach we go with. I'm also not aware of any specific examples of this (single interactive element plus extra content in a cell) for column types we currently know about.

- If we are just querying for specific element types in a cell (a single `nimble-anchor`/`nimble-button`/etc), I think we'd probably just go ahead and focus it if we find just one. I don't think we'd try to look for additional text in the cell and do something different if we find it.
- If we do have a new cell view API to handle the focusing behavior, I guess we'd just want some good documentation/guidance on when an element in the cell should get focused.

I did try out the NVDA screen reader some, and your reading of the ARIA guidelines seems accurate. If we focus an element inside a cell, that element (and it's label) will be read out loud, but any additional content in the cell not associated with that element will not be read. However there are some cases where it would be read (like the interactive element having an `aria-describedby` attribute that points to the other content).

So it's all a bit complicated, but I'm not sure we need to fully commit one way or another here (unless we expect this to come up for a known column type soon).